### PR TITLE
hackdayz#2: Doc fixes

### PR DESCRIPTION
**Pull request by clemens-fricke**
When building the doc multiple warnings are thrown. These should all be fixed in this PR.

This is done by adding mock imports for the required and optional packages.
Fixing documentation of namedtuples
Fix documentation formatting
Wrong typing (@clemens-fricke is was your fault...)

### CODE REVIEW

**Comment by clemens-fricke** [comment]
On Monday, I might look into adding pre-commit-hooks.

File: gustaf/helpers/data.py [hunk 1]
```diff
@@ -308,7 +308,7 @@ def __init__(self, helpee, **kwrags):
         Parameters
         -----------
         helpee: GustafBase
-        kwrags: **kwrags
+        kwrags: kwargs
```
Comment (j042) [inline gustaf/helpers/data.py:311]:
```suggestion
        **kwargs: kwargs
```

File: gustaf/vertices.py [hunk 2]
```diff
@@ -34,16 +34,12 @@ def __init__(
 
         Parameters
         -----------
+        whatami: str
```
Comment (j042) [inline gustaf/vertices.py:37]: `whatami` is not a param

File: gustaf/vertices.py [hunk 3]
```diff
@@ -34,16 +34,12 @@ def __init__(
...
         vertices: (n, d) np.ndarray
 
         Returns
         --------
         None
-
-        Attributes
-        -----------
-        whatami: str
-        vertices: np.ndarray
```
Comment (j042) [inline gustaf/vertices.py:46]: I guess this should come directly at class level. but not in scope of this pr

File: gustaf/utils/connec.py [hunk 4]
```diff
@@ -154,10 +154,10 @@ def faces_to_edges(faces):
...
         Ref: (node_ind), edge_ind
 
-             (0)
-             /\
-          0 /  \2
-           /____\
+            (0)
+             /\\
+          0 /  \\2
+           /____\\
```
Comment (j042) [inline gustaf/utils/connec.py:160]: this works?

File: gustaf/utils/arr.py [hunk 5]
```diff
@@ -106,10 +106,10 @@ def close_rows(arr, tolerance=None):
...
     --------
     close_stuff: tuple
-      unique_arrays: (n, d) np.ndarray
-      unique_ids: (m) np.ndarray
-      inverse: (n) np.ndarray
-      overlapping: list(list)
+    unique_arrays: (n, d) np.ndarray
+    unique_ids: (m) np.ndarray
+    inverse: (n) np.ndarray
+    overlapping: list(list)
```
Comment (j042) [inline gustaf/utils/arr.py:112]: I meant to state what's in the tuple, but I guess we can either rm `close_stuff` or just state tuple elems as a sentence

File: gustaf/helpers/raise_if.py [hunk 6]
```diff
@@ -10,17 +10,15 @@ def invalid_inherited_attr(func, qualname, property_=False):
...
-       behaves same as func if property_ is correctly defined
+    Args:
+        func (function): _description_
+        qualname (class): _description_
+        property_ (bool, optional):
+            is this function a property?. Defaults to False.
+
+    Returns:
+        raiser: function
+            behaves same as func if `property_` is correctly defined
```
Comment (j042) [inline gustaf/helpers/raise_if.py:21]: why is this a different format?

File: gustaf/helpers/data.py [hunk 7]
```diff
@@ -411,32 +413,27 @@ def compute_or_return_saved(*args, **kwargs):
...
-values: (n, d) np.ndarray
-ids: (n) np.ndarray
-inverse: (m) np.ndarray
-counts: (n) np.ndarray
 """
 
+Unique2DIntegers.values.__doc__ = "(n, d) np.ndarray"
+Unique2DIntegers.ids.__doc__ = "(n) np.ndarray"
+Unique2DIntegers.inverse.__doc__ = "(m) np.ndarray"
+Unique2DIntegers.counts.__doc__ = "(n) np.ndarray"
```
Comment (j042) [inline gustaf/helpers/data.py:435]: Can we give a hint, just like default doc, `field number 0` or something? if that gets too complicated, I am very okay with this one

**Review by j042 — COMMENTED** [review]
Thank you so much! I just have a few questions

File: gustaf/vertices.py (hunk 3 above)
Comment (clemens-fricke) [reply gustaf/vertices.py:46]: This should already be documented as the property is handled as decorated functions. Documentations should come from here.

File: gustaf/vertices.py (hunk 2 above)
Comment (clemens-fricke) [reply gustaf/vertices.py:37]: Removed

File: gustaf/helpers/data.py (hunk 1 above)
Comment (clemens-fricke) [reply gustaf/helpers/data.py:311]: Here you don't use the kwargs. Can it be removed, or do you need it?

File: gustaf/utils/arr.py (hunk 5 above)
Comment (clemens-fricke) [reply gustaf/utils/arr.py:112]: could we talk about this?

File: gustaf/utils/connec.py (hunk 4 above)
Comment (clemens-fricke) [reply gustaf/utils/connec.py:160]: yes

File: gustaf/helpers/data.py (hunk 1 above)
Comment (j042) [reply gustaf/helpers/data.py:311]: Nice catch, thanks,  it can be removed

**Review by j042 — APPROVED** [review]
perfect, thank you so much! now we can add doc check to CI!

**Pull request merged**